File: backjoon/완전탐색/Python/블랙잭.py
def solve(N, K, Array) :

    sum = 0

    for i in range(0, N) :
        for j in range(i+1, N) :
            for k in range(j+1, N) :
                if(Array[i] + Array[j] + Array[k] > K) :
                    continue
                else :
                    sum = max(sum, Array[i] + Array[j] + Array[k])

    print(sum)



if __name__ == "__main__":
    N, K = list(map(int, input().split()))
    Array = list(map(int, input().split()))
    solve(N, K, Array)  

File: backjoon/완전탐색/JS/블랙잭.js
const input = require("fs")
  .readFileSync(process.platform === "linux" ? "./dev/stdin" : "./input.txt")
  .toString()
  .trim()
  .split("\n");

const [N, K] = input.shift().split(" ").map(Number);

const Array = input[0].split(" ").map(Number);
let sum = 0;

for (let i = 0; i < N; i++) {
  for (let j = i + 1; j < N; j++) {
    for (let k = j + 1; k < N; k++) {
      if (Array[i] + Array[j] + Array[k] > K) continue;
      else {
        sum = Math.max(sum, Array[i] + Array[j] + Array[k]);
      }
    }
  }
}

console.log(sum);
